index.js

```diff
@@ -1,312 +1 @@
-    data: {
-      title: options.title,
-      content: options.content,
-      contentFormat: options.contentFormat,
-      tags: options.tags,
-      canonicalUrl: options.canonicalUrl,
-      publishStatus: options.publishStatus,
-      license: options.license
-    }
-  }, callback)
-}
-
-
-/**
- * Acquires an access token for the Medium API.
- *
- * Sets the access token on the client on success.
- *
- * @param {Object} params
- * @param {NodeCallback} callback
- */
-MediumClient.prototype._acquireAccessToken = function (params, callback) {
-  this._makeRequest({
-    method: 'POST',
-    path: '/v1/tokens',
-    contentType: 'application/x-www-form-urlencoded',
-    data: qs.stringify(params)
-  }, function (err, data) {
-    if (!err) {
-      this._accessToken = data.accessToken
-    }
-    callback(err, data)
-  }.bind(this))
-}
-
-
-/**
- * Makes a request to the Medium API.
- *
- * @param {Object} options
- * @param {NodeCallback} callback
- */
-MediumClient.prototype._makeRequest = function (options, callback) {
-  var requestParams = {
-    host: 'api.medium.com',
-    port: 443,
-    method: options.method,
-    path: options.path
-  }
-  var req = https.request(requestParams, function (res) {
-    var body = []
-
-    res.setEncoding('utf-8')
-    res.on('data', function (data) {
-      body.push(data)
-    })
-    res.on('end', function () {
-      var payload
-      var responseText = body.join('')
-      try {
-        payload = JSON.parse(responseText)
-      } catch (err) {
-        callback(new MediumError('Failed to parse response', DEFAULT_ERROR_CODE), null)
-        return
-      }
-
-      var statusCode = res.statusCode
-      var statusType = Math.floor(res.statusCode / 100)
-
-      if (statusType == 4 || statusType == 5) {
-        var err = payload.errors[0]
-        callback(new MediumError(err.message, err.code), null)
-      } else if (statusType == 2) {
-        callback(null, payload.data)
-      } else {
-        callback(new MediumError('Unexpected response', DEFAULT_ERROR_CODE), null)
-      }
-    })
-  }).on('error', function (err) {
-    callback(new MediumError(err.message, DEFAULT_ERROR_CODE), null)
-  })
-
-  req.setHeader('Content-Type', options.contentType || 'application/json')
-  req.setHeader('Authorization', 'Bearer ' + this._accessToken)
-  req.setHeader('Accept', 'application/json')
-  req.setHeader('Accept-Charset', 'utf-8')
-
-  req.setTimeout(DEFAULT_TIMEOUT_MS, function () {
-    // Aborting a request triggers the 'error' event.
-    req.abort()
-  })
-
-  if (options.data) {
-    var data = options.data
-    if (typeof data == 'object') {
-      data = JSON.stringify(data)
-    }
-    req.write(data)
-  }
-  req.end()
-}
-
-// Exports
-
-module.exports = {
-  MediumClient: MediumClient,
-  Scope: Scope,
-  PostPublishStatus: PostPublishStatus,
-  PostLicense: PostLicense,
-  PostContentFormat: PostContentFormat
-}
+module.exports = require("./lib/medium-client.js")
```

Comment:
With adding tests, felt necessary to pull actual code into lib/, which makes this pull request harder to review. I'll add notes on medium-client.js to show where I changed things.